File: TransCoder_tokenized_test_set_functions/cpp/FIND_LARGEST_D_IN_ARRAY_SUCH_THAT_A_B_C_D_1.cpp
int findFourElements ( int arr [ ], int n ) {
  unordered_map < int, pair < int, int > > mp;
  for ( int i = 0;
  i < n - 1;
  i ++ ) for ( int j = i + 1;
  j < n;
  j ++ ) mp [ arr [ i ] + arr [ j ] ] = {
    i, j };
    int d = INT_MIN;
    for ( int i = 0;
    i < n - 1;
    i ++ ) {
      for ( int j = i + 1;
      j < n;
      j ++ ) {
        int abs_diff = abs ( arr [ i ] - arr [ j ] );
        if ( mp . find ( abs_diff ) != mp . end ( ) ) {
          pair < int, int > p = mp [ abs_diff ];
          if ( p . first != i && p . first != j && p . second != i && p . second != j ) d = max ( d, max ( arr [ i ], arr [ j ] ) );
        }
      }
    }
    return d;
  }

File: TransCoder_tokenized_test_set_functions/java/FIND_LARGEST_D_IN_ARRAY_SUCH_THAT_A_B_C_D_1.java
static int findFourElements ( int [ ] arr, int n ) {
  HashMap < Integer, Indexes > map = new HashMap < > ( );
  for ( int i = 0;
  i < n - 1;
  i ++ ) {
    for ( int j = i + 1;
    j < n;
    j ++ ) {
      map . put ( arr [ i ] + arr [ j ], new Indexes ( i, j ) );
    }
  }
  int d = Integer . MIN_VALUE;
  for ( int i = 0;
  i < n - 1;
  i ++ ) {
    for ( int j = i + 1;
    j < n;
    j ++ ) {
      int abs_diff = Math . abs ( arr [ i ] - arr [ j ] );
      if ( map . containsKey ( abs_diff ) ) {
        Indexes indexes = map . get ( abs_diff );
        if ( indexes . getI ( ) != i && indexes . getI ( ) != j && indexes . getJ ( ) != i && indexes . getJ ( ) != j ) {
          d = Math . max ( d, Math . max ( arr [ i ], arr [ j ] ) );
        }
      }
    }
  }
  return d;
}